clients/java/zms/src/main/java/com/yahoo/athenz/zms/ZMSClient.java

```diff
@@ -2861,17 +2861,18 @@ public void putMembershipDecision(String domainName, String roleName, String mem
     }
 
     /**
-     * Return all the list of pending requests for the given principal. If the principal
+     * Return all the list of pending requests for the given principal or domain. If the principal
      * is null, the server will return the list for the authenticated principal
      * making the call
      * @param principal name of the approver principal (optional)
+     * @param domainName name of the domain to get pending requests for (optional)
      * @return DomainRoleMembership object listing all pending users
      * @throws ZMSClientException in case of failure
      */
-    public DomainRoleMembership getPendingDomainRoleMembersList(String principal) {
+    public DomainRoleMembership getPendingDomainRoleMembersList(String principal, String domainName) {
```

Comment:
we can't change existing APIs as that will introduce a backward incompatible version.

instead add a new api with the new argument and then update the existing one to call with a null argument. e.g. take a look at the getMembership api call.

Reply:
fixed